app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -96,58 +96,78 @@ class WallpapersUseCases(
             // This should be cleaned up as improvements are made to the storage, file management,
             // and download utilities.
             withContext(Dispatchers.IO) {
-                val availableWallpapers = getAvailableWallpapers()
+                val availableWallpapers = possibleWallpapers.getAvailableWallpapers()
                 val currentWallpaperName = settings.currentWallpaper
                 val currentWallpaper = possibleWallpapers.find { it.name == currentWallpaperName }
                     ?: fileManager.lookupExpiredWallpaper(currentWallpaperName)
                     ?: Wallpaper.Default
 
                 fileManager.clean(
                     currentWallpaper,
-                    possibleWallpapers.filterIsInstance<Wallpaper.Remote>()
+                    possibleWallpapers
                 )
                 downloadAllRemoteWallpapers(availableWallpapers)
                 store.dispatch(AppAction.WallpaperAction.UpdateAvailableWallpapers(availableWallpapers))
                 store.dispatch(AppAction.WallpaperAction.UpdateCurrentWallpaper(currentWallpaper))
             }
         }
 
-        private fun getAvailableWallpapers() = possibleWallpapers
-            .filter { !it.isExpired() && it.isAvailableInLocale() }
+        private fun List<Wallpaper>.getAvailableWallpapers() =
+            this.filter { !it.isExpired() && it.isAvailableInLocale() }
 
         private suspend fun downloadAllRemoteWallpapers(availableWallpapers: List<Wallpaper>) {
-            for (wallpaper in availableWallpapers.filterIsInstance<Wallpaper.Remote>()) {
+            for (wallpaper in availableWallpapers) {
+                if (wallpaper == Wallpaper.Default) continue
                 downloader.downloadWallpaper(wallpaper)
```

Comment:
personal nit: sometimes the `continue` keyword can be make a logic flow a little ambiguous; I usually prefer to avoid it, if possible. 
```suggestion
                if (wallpaper != Wallpaper.Default) {
                    downloader.downloadWallpaper(wallpaper)
                }
```

Reply:
What would hypothetically happen here if `Wallpaper.Default` was passed into `downloader.downloadWallpaper`

Reply:
I can dig your syntax suggestion, will update.

I had thought that the downloader would error if it couldn't find a file (which is wouldn't be able to for the default), but it looks like it just silently fails. I would be okay with reverting that specific line as well, but some of the logic here will probably be reworking anyway once the `possibleWallpapers` are all being fetched remotely.